mistralrs-core/src/pipeline/gguf.rs

```diff
@@ -657,8 +583,8 @@ impl Pipeline for GGUFPipeline {
         }
     }
     fn activate_adapters(&mut self, adapter_names: Vec<String>) -> anyhow::Result<usize> {
-        if !self.metadata.is_lora {
-            anyhow::bail!("Cannot activate adapters non-LoRA models.")
+        if !self.metadata.has_adapter {
```

Comment:
Ok, could we better communicate this?

LoRA models share the same as X-LoRA when they're created? GGUF/GGML is converted into `XLoraLlama` for example, but this has an `is_xlora() -> true` method.

Something seems off, or for someone like me not too familiar with these terms, other than you were involved in developing X-LoRA, I don't know how the usage differs in respect to `XLoraLlama` vs `GeneralMetadata` since `is_xlora` and `is_xlora()` kind of conflict there, but the model and pipeline are presumably related?

---

Is `activate_adapters()` only meant for LoRA? Or for any potential `AdapterKind` that isn't X-LoRA?

```rs
// Pseudo-code of what might be a better way to query?
if self.metadata.adapter.is_some_and(|a| !a.is_xlora()) { activate } else { bail }
```